### PR TITLE
WIP plugin/kubernetes: Allow multiple Kubernetes

### PR DESCRIPTION
**Pull request by chrisohaver**
### 1. What does this pull request do?

Allows multiple kubernetes blocks to be defined.

### 2. Which issues (if any) are related?

Implements #1483 

### 3. Which documentation changes (if any) need to be made?

Excellent question. Should we document this?  We do not mention it for other plugins that support multiple declarations.

### CODE REVIEW

**Comment by chrisohaver** [comment]
adding an integration test: /integration-cipr35

File: plugin/kubernetes/federation.go [hunk 1]
```diff
@@ -23,23 +24,30 @@ const (
...
-		return msg.Service{Host: dnsutil.Join([]string{r.service, r.namespace, fname, r.podOrSvc, lz, lr, fzone})}, nil
-	}
+		if r.endpoint == "" {
+			return msg.Service{Host: dnsutil.Join([]string{r.service, r.namespace, fname, r.podOrSvc, lz, lr, fzone})}, nil
+		}
 
-	return msg.Service{Host: dnsutil.Join([]string{r.endpoint, r.service, r.namespace, fname, r.podOrSvc, lz, lr, fzone})}, nil
+		return msg.Service{Host: dnsutil.Join([]string{r.endpoint, r.service, r.namespace, fname, r.podOrSvc, lz, lr, fzone})}, nil
+	}
+	return msg.Service{}, nil // need to return error here?
```
Comment (miekg) [inline plugin/kubernetes/federation.go:52]: we only returned an empty service with `err != nil`, so I think we should return an error here

**Review by miekg — COMMENTED** [review]
look sane.

Would rename Hander to Kubernetes and name the []*Kubernetes differently. Would mention this in the README.md that we *explicitly* allow this. Also haven't seen @costinm ack/nack anything.

Is this looks and feels awfully similar to the (failed) federation I'm not to tricky happy to merge this.

**Comment by chrisohaver** [comment]
When writing new integration tests(coredns/ci#35), I discovered this change isn't quite working as I thought...  Need to dive in deeper to see why.

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/coredns] plugin/kubern..." ]
>/integration-cipr35

an of the bat concern: monitoring and metrics?

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [coredns/coredns] WIP plugin/ku..." ]
>When writing new integration tests(coredns/ci#35), I discovered this change isn't quite working as I thought...  Need to dive in deeper to see why.

I wouldn't spend too many cycles on this. You're diluting kubernetes for a use 
case that <0.01% of our users have. I rather see any hackery done in a istio 
plugin (or at least explore that possibility)

**Comment by chrisohaver** [comment]
Closing this PR. Will move to an external plugin.

**Pull request closed without merging**